README.md: Add sections for bash/zsh completion (#75)
lhark/truckersmp-cli: README.md

$ make
```

## bash/zsh completion

If [`genzshcomp`][python-genzshcomp] is installed, `make` generates shell completion files for bash (bash-completion) and zsh.

> They enable completion of available options.

Reviewer: ```suggestion
They enable tab completions of available command line options.
```

Author: Googled but "tab completion*s* of available" was not found.

### System-wide installation

Shell|System-wide search paths
---|---
bash|`$(pkg-config --variable=completionsdir bash-completion)` (e.g. `/usr/share/bash-completion/completions/`)
zsh|`/usr/share/zsh/site-functions/`, `/usr/local/share/zsh/site-functions/`

> * bash-completion file `truckersmp-cli.bash` needs to be renamed to `truckersmp-cli`

Reviewer: Should we provide an install target in the makefile which moves these files? What do you think?

Author: Where is the *unique* destination directory? I think it's difficult to provide it.

Reviewer: Probably `$prefix/share/{zsh,bash-completion}` with defaulting `prefix` to `/usr/local`?
Let's skip this.

Author: I tested `/usr/local/share/bash-completion/completions/` and it's working.

So we can use the following directories:

* bash: `/usr/local/share/bash-completion/completions/`
* zsh: `/usr/local/share/zsh/site-functions/`

However I'm not sure whether another location (prefix) can be used because zsh especially only supports `/usr/local/share/zsh/` for manually installed function files:

```c
/* From zsh-5.8/Src/init.c */

# ifdef FIXED_FPATH_DIR
    /* Zeroth: /usr/local/share/zsh/site-functions */
    *fpathptr++ = ztrdup(FIXED_FPATH_DIR);
    fpathlen--;
# endif
# ifdef SITEFPATH_DIR
    /* First: the directory from --enable-site-fndir
     *
     * default: /usr/local/share/zsh/site-functions
     * (but changeable by passing --prefix or --datadir to configure) */
    *fpathptr++ = ztrdup(SITEFPATH_DIR);
    fpathlen--;
# endif /* SITEFPATH_DIR */
# if defined(ADDITIONAL_FPATH)
    /* Second: the directories from --enable-additional-fpath
     * 
     * default: empty list */
    for (j = 0; j < more_fndirs_len; j++)
	*fpathptr++ = ztrdup(more_fndirs[j]);
# endif
# ifdef FPATH_DIR
    /* Third: The directory from --enable-fndir
     *
     * default: /usr/local/share/zsh/${ZSH_VERSION}/functions */
#  ifdef FPATH_SUBDIRS
#   ifdef ADDITIONAL_FPATH
    for (j = more_fndirs_len; j < fpathlen; j++)
	*fpathptr++ = tricat(FPATH_DIR, "/", fpath_subdirs[j - more_fndirs_len]);
#   else
    for (j = 0; j < fpathlen; j++)
	*fpathptr++ = tricat(FPATH_DIR, "/", fpath_subdirs[j]);
#   endif
#  else
    *fpathptr++ = ztrdup(FPATH_DIR);
#  endif
# endif
```

> * bash-completion file `truckersmp-cli.bash` needs to be renamed to `truckersmp-cli`

Reviewer: ```suggestion
* The bash-completion file `truckersmp-cli.bash` needs to be renamed to `truckersmp-cli`
```

> * Debian-based systems have `/usr/share/zsh/vendor-completions/` directory for zsh completions

Reviewer: ```suggestion
* Debian-based systems are using the `/usr/share/zsh/vendor-completions/` directory for zsh completions
```

### Per-user installation

#### bash

Copy `truckersmp-cli.bash` to `$XDG_DATA_HOME/bash-completion/completions/truckersmp-cli`.

```

> $ mkdir -p ${XDG_DATA_HOME:-~/.local/share}/bash-completion/completions
> $ cp truckersmp-cli.bash ${XDG_DATA_HOME:-~/.local/share}/bash-completion/completions/truckersmp-cli

Reviewer: Paths can contain whitespaces so we're better off with quoting them.

```suggestion
$ mkdir -p "${XDG_DATA_HOME:-~/.local/share}/bash-completion/completions"
$ cp truckersmp-cli.bash "${XDG_DATA_HOME:-~/.local/share}/bash-completion/completions/truckersmp-cli"
```

```

#### zsh

> Copy `_truckersmp-cli` to a directory that is part of `$fpath`.

Reviewer: ```suggestion
Copy `_truckersmp-cli` to a directory that is part of `$fpath` and run `compinit`.
```

```zsh
# Example: `_truckersmp-cli` is already placed in ~/.zsh.d/comp/

> fpath=(~/.zsh.d/comp/ $fpath)

Reviewer: We already expect the file in the `$fpath`.

autoload -U ~/.zsh.d/comp/*(:t)

> autoload -Uz compinit && compinit

Reviewer: When the file is in the `$fpath` a simple `compinit` should be enough.

```

> 

Reviewer: I would make this a subsection of _Install_.
